src/operations/query/filters/string.js

```diff
@@ -9,26 +11,65 @@
 function filterByString(queryParameterValue, and_segments, propertyObj, columns) {
     if (Array.isArray(queryParameterValue)) {
         // if array is passed then check in array
-        and_segments.push({
-            [`${propertyObj.field}`]: {
-                $in: queryParameterValue,
-            },
-        });
+
+        if (`${propertyObj.fields}`) {
+            and_segments.push({
+                $or: propertyObj.fields.map((f) => {
+                    return {
+                        [`${f}`]: {
+                            $in: queryParameterValue}
+                    };
+                }),
+            });
+            columns.add(`${propertyObj.fields}`);
+        }
+        else {
+            and_segments.push({
+                [`${propertyObj.field}`]: {
+                    $in: queryParameterValue,
+                },
+            });
+            columns.add(`${propertyObj.field}`);
```

Comment:
Could move the 'columns.add' call outside of the if else since its called in both if/else parts

Reply:
No, the call is slightly different. One is looking at the field key, the other is looking at the field**s** key. I could take this out and do an if/else looking at the field key. That would be more elegant coding, consolidating the action, but it would also add an unnecessary if/else block slightly decreasing performance.